include/cocaine/api/executor.hpp

```diff
@@ -6,6 +6,20 @@ namespace cocaine {
 namespace api {
 
 class executor_t {
+    template<class F>
+    class move_only_wrapper {
+    public:
+        move_only_wrapper(F _f):
+            f(new F(std::move(_f)))
+        {}
+
+        auto operator()() -> void {
```

Comment:
not really sure) - https://github.com/3Hren/cocaine-core/blob/v0.12.11/src/service/runtime.cpp#L27